### PR TITLE
Feature/111/create recipe form

### PR DESCRIPTION
**Pull request by mglavey**
### Describe The Problem Being Solved:
Create recipe form.

### Checklist For Submitter

* [ ] I have documented all new functionality
* [ ] I have screenshotted new UI changes and attached them to this PR

### Checklist For Reviewer
* [ ] Code formatting/static analysis: is the code formatted properly? Does a linter/other form of static analysis reveal any issues? Does the code have appropriately low cyclomatic complexity?
* [ ] Code architecture: concerns are separated, code follows single-responsibility principle, code utilizes OOP, DRY code.
* [ ] Coding best practices: code avoids hard-coded variables, inefficient computations, reinventing the wheel when using frameworks. Code is well-commented and generally readable.
* [ ] Non-functional requirements: Code has accompanying tests (if using TDD). Programmer can easily explain decisions to reviewers.

### CODE REVIEW

**Comment by mglavey** [comment]
Hi folks,
This form is by no means done, but since docker is acting up again and it's been a little slow, I thought I'd try an 'in progress' code review, just to look over what I already have and see if we can make any adjustments before finishing the whole thing out. Let me know what you think.

File: frontend/src/components/CreateRecipeForm/CreateRecipeForm.js [hunk 1]
```diff
@@ -96,11 +98,11 @@ class CreateRecipeForm extends Component {
...
-              <input class="form-control" type="text" placeholder=""></input>
-              <input class="form-control" type="text" placeholder=""></input>
-              <input class="form-control" type="text" placeholder=""></input>
-              <input class="form-control" type="text" placeholder=""></input>
-              <input class="form-control" type="text" placeholder=""></input>
+              <input class="form-control" type="text" placeholder="" ></input>
+              <input class="form-control" type="text" placeholder="" ></input>
+              <input class="form-control" type="text" placeholder="" ></input>
+              <input class="form-control" type="text" placeholder="" ></input>
+              <input class="form-control" type="text" placeholder="" ></input>
```
Comment (vhoof) [inline frontend/src/components/CreateRecipeForm/CreateRecipeForm.js]: This kind of a question of taste, but I'm not sure we're using these before-closing spaces anywhere else. I personally find them odd, but maybe that's just my HTML1 biases showing through.

**Comment by mglavey** [comment]
image
image
image

File: frontend/src/components/EditRecipe/EditRecipe.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+          </div>
+        </div>
+        <div className='row form-photo-upload'>
+          <div className='col-6 offset-3'>
+            <ImageUploader 
+              withIcon={true}
+              buttonText='choose images'
+              onChange={this.onDrop}
+              imgExtension={[ '.jpg', '.gif', '.png' ]}
+              maxFileSize={5242880}
```
Comment (vhoof) [inline frontend/src/components/EditRecipe/EditRecipe.js]: So I think the goal here is less to have 5 different places to upload, and more to have one upload box and display the pics that have already been uploaded below it. The colors, buttons, and dotted line need a few style changes.

File: frontend/src/components/EditRecipe/EditRecipe.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+                  <div className='direction-input'><input className='form-control' type='text' placeholder=''></input></div>
+                  <div className='direction-input'><input className='form-control' type='text' placeholder=''></input></div> 
+                </div>
+              </div>
+              <div className='col-1'>
+                <div className='row delete-input'><i class="fas fa-times"></i></div>
+                <div className='row delete-input'><i class="fas fa-times"></i></div>
+                <div className='row delete-input'><i class="fas fa-times"></i></div>
+                <div className='row delete-input'><i class="fas fa-times"></i></div>
+                <div className='row delete-input'><i class="fas fa-times"></i></div>
```
Comment (vhoof) [inline frontend/src/components/EditRecipe/EditRecipe.js]: I don't want to be a pixel-perfect lunatic, but I think these could be a bit bigger.

File: frontend/src/components/EditRecipe/EditRecipe.js [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+          </div>
+        </div>
+        <div className='row form-tag-input'>
+          <div className='col-4 offset-4'>
+            <div className='row tag-input'>
+              <div className='col-11'>
+                <div className='form-group'>
+                  <label for='tag'><h2>Tags</h2></label>
+                  <input className='form-control' type='text' placeholder='Search and tag'></input>          
+                  <button className='card-meal-btn'>DINNER</button>
```
[the remark this answers is not in the document]
Comment (vhoof) [reply frontend/src/components/EditRecipe/EditRecipe.js]: I think we also want the search symbol inside the input bar. #181 has one possible implementation there. And we may want to use the same search icon in both places for consistency.

**Comment by vhoof** [comment]
I think there are some horizontal rules that are also present in the design between elements.

File: frontend/src/components/EditRecipe/EditRecipe.js [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+            <div>Ingredients</div>
+          </div>
+        </div>
+        <div className='row'>
+          <div className='col-8 offset-2'>
+            <div className='row form-ingredients'>
+              <div className='col-7 offset-2'>
+                <div className='ingredients form-group'>
+                  <div className='ingredient-input'><input className='form-control' type='text' placeholder=''></input></div>
+                  <div className='ingredient-input'><input className='form-control' type='text' placeholder=''></input></div>
```
[the remark this answers is not in the document]
Comment (vhoof) [reply frontend/src/components/EditRecipe/EditRecipe.js]: Looking at it again, we may also want to use the brighter yellow (like the Donate button above) and have it shift to the darker on hover. But if that doesn't look right, feel free to leave it as is.

**Comment by vhoof** [comment]
@mglavey Were you planning to break up the remaining work into smaller, iterative PRs, or continue working off this one? If it's the former, it might make sense to close this for now, and revisit each of the pieces separately.

**Comment by mglavey** [comment]
Hey Bryan,

I think it's best to break this PR up iteratively and work on the separate parts. I've started researching Base64 photouploading, so that can be the first ticket I'm working on. I'll create subtickets on the 111/Create Recipe card on Zenhub, that way I can more definitively track the progress of the Create Recipe form overall.

We can go ahead and close this PR.

**Pull request closed without merging**